src/borg/archive.py

```diff
@@ -291,7 +291,9 @@ def __init__(self, repository, key, manifest, name, cache=None, create=False,
         self.hard_links = {}
         self.stats = Statistics(output_json=log_json)
         self.show_progress = progress
-        self.name = name
+        self.name = name  # overwritten later with name from archive metadata
+        self.name_in_manifest = name  # can differ from .name later (if borg check fixed duplicate archive names)
```

Comment:
yes, i noticed also it is not actively used.

i wanted to keep it as it points out the issue with different names in manifest and in archive.